### DISCUSSION TITLE
Getting thermal efficiency of a shell and tube heat exchanger with a Python Script

### DISCUSSION THREAD

**Discussion by igors96** [DanWBR/dwsim#437] [General]
Hi, Dan!

I'm being introduced to use Python Scripts in my DWSIM projects. I've worked on a simple project of a shell and tube heat exchanger, and I wanna collect the thermal efficiency of many simulations fastly. This is the basic structure of the flowsheet:

![image](https://user-images.githubusercontent.com/67437213/221264069-63b3484f-c63f-4363-ada3-e4d4f2b33c46.png)

I have a Excel file with many combinations of temperatures and flows of hot and cold fluid, and I'm studying too about the influence of the shell passes in the thermal efficiency. So, the Excel file is something like this:

![image](https://user-images.githubusercontent.com/67437213/221265698-0c8bb220-7a85-4ff1-ad7c-bde3a5ea6b8f.png)

How can I make these simulations in DWSIM with a Python Script, in order to collect the thermal efficiency, and export the data to a Excel or CSV file? Is it possible?

Thank you!

**Comment by DanWBR** [comment]
You can copy the data to the spreadsheet and access the cells with

`cellvalue = Spreadsheet.Worksheets[0].Cells[i, j].Data
`

Then put this inside the loop:

```
hotstream = Flowsheet.GetObject('Hot Water Inlet')
coldstream = Flowsheet.GetObject('Cold Water Inlet')

hotstream.SetTemperature(Thot)
coldstream.SetTemperature(Tcold)

hx = Flowsheet.GetObject('HE-01')
hx.STProperties.Shell_NumberOfPasses = npasses

Flowsheet.RequestCalculation()

efficiency = hx.ThermalEfficiency
```

**Comment by virajdesai0309** [comment]
You may import the flowsheet in the python environment using. 
LoadFlowsheet() method and then can target the mass flow of cold inlet stream for a range using for loop and get the thermal efficiency of HEX on to the terminal and export the data to csv using python pandas
